apps/arena/lib/arena/game/skill.ex

```diff
@@ -196,6 +202,22 @@ defmodule Arena.Game.Skill do
       |> Map.put(:is_moving, true)
       |> Map.put(:speed, speed)
       |> put_in([:aditional_info, :forced_movement], true)
+      |> maybe_make_invinsible(leap)
+
+    put_in(game_state, [:players, player.id], player)
+  end
+
+  def do_mechanic(game_state, player, {:teleport, teleport}, %{skill_target: skill_target}) do
+    target_position = %{
+      x: player.position.x + skill_target.x * teleport.range,
+      y: player.position.y + skill_target.y * teleport.range
+    }
+
+    player =
+      player
+      |> Physics.move_entity_to_position(target_position, game_state.external_wall)
+      |> Map.put(:aditional_info, player.aditional_info)
+      |> maybe_make_invinsible(teleport)
```

Comment:
Just to make sure I understand, they idea is to give you 300 ms (current config) of damage immunity **after** you teleport?

Reply:
Then why do you need the immunity? Teleportation is instant, how could some damage you while teleporting?

Reply:
That's a hacky solution for how the teleport will be represented in the front, they will make the character model to go under the floor, walk down there and then get up on the final destinations. Thinking about it, maybe we don't even need a new case for `do_mechanic` because we could simply make it a dash and call it a teleport, the thing is that the character won't be visible, so the invisibility is needed

Reply:
JAJAJAJA that explains it

🤔 mmm honestly don't like any of those hacks... would it be possible for them to dissapear and appear the model? We could, in theory, remove the player from `game_state.players` and bring it back say 30 ms laters (or whatever we configure), essentially disappear the player for some time